dropshot/src/http_util.rs

```diff
@@ -35,9 +36,11 @@ where
     // TODO better understand pin_mut!()
     // TODO do we need to use saturating_add() here?
     let mut nbytesread: usize = 0;
-    while let Some(maybebuf) = body.data().await {
-        let buf = maybebuf?;
-        nbytesread += buf.len();
+    while let Some(maybefr) = body.frame().await {
+        let fr = maybefr?;
+        if let Ok(buf) = fr.into_data() {
+            nbytesread += buf.len();
+        }
```

Comment:
AFAICT, this code will eat any `TRAILERS` frame, which the previous implementation wouldn't do. This is actually probably *more* correct given what this code is intended to do.